lib/template-loader.php

```diff
@@ -323,11 +246,11 @@ function gutenberg_find_template_post_and_parts( $template_type, $template_hiera
 
 	if ( $current_template_post ) {
 		$template_part_ids = array();
-		if ( is_admin() || defined( 'REST_REQUEST' ) ) {
+		/* if ( is_admin() || defined( 'REST_REQUEST' ) ) {
 			foreach ( parse_blocks( $current_template_post->post_content ) as $block ) {
 				$template_part_ids = array_merge( $template_part_ids, create_auto_draft_for_template_part_block( $block ) );
 			}
-		}
+		} */
```

Comment:
This is such a weird thing to do in a function called `gutenberg_find_template_post_and_parts`. I'm not sure I understand yet, what impact this change has for the moment.

Reply:
Im not entirely sure either.

One thing is that the template at this point may just reference the template parts by slug/theme names.  And the `create_auto_draft_for_template_part_block` function runs to determine the ids of the entities (not necessarily to create them) so their content can properly be displayed?  The block in edit does this on its own, but maybe this is necessary for the front-end?  🤔

Reply:
I agree with Addie, and here's my chain of reasoning for this code:

#### The function `create_auto_draft_for_template_part_block` does the following:
1. Searches the given post content for a template part block.
2. Given the template part block's slug/theme, finds the template part .html file in the filesystem.
3. Creates a post for that template part if it doesn't exist yet and saves the id, OR if the post exists, saves the id of the existing post.
4. Returns a map of "template part slug" to "template part IDs"

#### The function `gutenberg_find_template_post_and_parts` does the following:
1. Searches for the correct hierarchical template to use for the given template type.
2. If a template post exists, uses that ID. If a template post does not exist, find the html file, and insert it as a post, and then use that ID.
3. Given the template post from above, search its content for template part blocks using `create_auto_draft_for_template_part_block`.
4. Returns the current template post, and all of the post IDs for each template part the template references.

### Summary of the behavior:
So basically, the idea is to pass in the template you need to resolve (e.g. `singular`), and then this code will make sure that the template _and_ all of the template parts referenced by this template are initialized into WordPress as autodrafts so that they can be used by the site editor.

By removing this code, it basically changes the function to _just_ resolve the template. So I think if we want to remove this code from here, we need to make sure that each template part block inside of that template has a corresponding template part CPT at some other point.

### Where is this behavior used:
Searching through the codebase, I see that functions use `gutenberg_find_template_post_and_parts` to get the template part data in the following cases:
1. In theme export, which relies on the returned template part IDs to export each of the template part CPTs. (So we would need a different way to get all of the template part IDs referenced by the template here.)
2. In `filter_rest_wp_template_part_query`, we rely on the function to give us all of the template part IDs associated with the template type. (E.g. we might request for "singular" template, and so the function gives us all the template part IDs which are associated with that template.) Then the query makes that the template parts it resolves are part of the IDs in the "singular" template. (So we would need a different way to determine which template parts are associated with a given template type.)

To the best of my knowledge, the other places where the function is used (`filter_rest_wp_template_query` and `gutenberg_override_query_template`) does not rely on this functionality.

I hope that's helpful!

Reply:
It looks like you also removed the code referenced in item 2 above, so I guess it's just used for theme export if that's not important any more!